src/Workspaces/Remote/ServiceHub/Host/RemoteWorkspace.SolutionCreator.cs

```diff
@@ -462,6 +503,17 @@ private async Task<Project> UpdateProjectInfoAsync(Project project, Checksum inf
                     await _assetProvider.SynchronizeProjectAssetsAsync(projectChecksums, cancellationToken).ConfigureAwait(false);
                 }
 
+                return await UpdateDocumentsAsync(project, addDocuments, removeDocuments, oldDocumentIdToStateChecksums, newDocumentIdToStateChecksums, cancellationToken).ConfigureAwait(false);
```

Comment:
extracted into helper (like we do with projects).  helps keep the syncing of checksums distinct from teh updating of state.